core/payment/src/lib.rs

```diff
@@ -73,3 +75,30 @@ impl PaymentService {
         log::info!("Payment service stopped.");
     }
 }
+
+async fn release_allocation(db: DbExecutor) {
+    if let Ok(allocations) = db
+        .as_dao::<dao::AllocationDao>()
+        .get_filtered(None, None, None, None, None)
+        .await
+    {
+        allocations.iter().for_each(|a| {
+            let allocation = a.clone();
+            let db = db.clone();
+            let deadline = allocation.timeout.clone().unwrap_or(Utc::now());
+            tokio::task::spawn_local(async move {
+                let deadline = deadline - Utc::now();
+                tokio::time::delay_for(deadline.to_std().unwrap_or(Duration::from_secs(0))).await;
```

Comment:
Please propose solutions, what should we do in case of conversion failure

Reply:
Other mechanism for deadline calculation has been implemented. This should be now bug-resistant.